### ISSUE TITLE
Workflow of ZfrRest

### ISSUE THREAD

**Issue by bakura10** [zf-fr/zfr-rest#29]
Hi,

As it is pretty confused for me and, it looks like, for you a little bit too, here I attempted to summarise the various steps of ZfrRest:
1. First step is to execute the Resource Route on MvcEvent::EVENT_ROUTE listener.
   1. If ResourceRoute returns null, an error will be triggered, end of process.
   2. Otherwise, returns a RouteMatch with the association resource and controller, fetched from the resource metadata.
2. Then, the MvcEvent::EVENT_DISPATCH is triggered. We have some listeners:
   1. The optional HttpMethodOverrideListener, that optionally override the Http method.
   2. Next I'd want to introduce a new listener, called "ProcessResourceListener" or something like that, whose job is to perform the various tasks like hydrating, filtering, body processing.
      1. If method is GET/DELETE: just leave the resource as it is in the route match. Methods in the controller will simply be public function get(User $user) / delete(User $user).
      2. If method is PUT/POST:
         1. Use the Body Parser to extract data (once again the question here is do we need to use the Body Parser, that only use the DecoderPluginManager, or should we use the decoders included in the resource... your use case you told me yesterday really looked weird to me so I don't really understand... The goal is to avoid asking people to write 30 decoders, 30 encodes, 30 hydrators... otherwise this module becomes pretty useless if we ask people to write so much boilerplate code).
         2. Get the input filter from the metadata and run it through the data fetched in previous step. If data is invalid, throw an exception.
         3. If data is valid, get the hydrator from the resource metadata, and hydrate the entity. Update the resource in the RouteMatch with what is returned from the hydrator.
         4. This will then resume, and the controller will be simply put(User $user) / post(User $user).
   3. The controller is run and returns "something".
   4. The SelectModelListener is called, which takes the result and parses to Json using the the resource encoder of the resource.

Some things here:
   Basically, we're not using the the DecoderPluginManager and EncoderPluginManager anymore, simply because after all, we are using the decoders/encoders specified in the resource metadata.

   What's still very unclear for me is, if for PUT/POST request, we should do all the stuff (filtering data, hydrating...) or if we should simply let people do this manually. If we let people do this manually this means that we can remove inputFilter and hydrator from the resource metadata.
